cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -332,22 +348,20 @@ func (d *Daemon) mustInitializeStorage(cfg *config.Config) *feewindow.FeeWindows
 				}).Debug("still initializing in-memory store")
 			}
 
-			if err := feewindows.IngestFees(txmeta); err != nil {
-				d.logger.WithError(err).Fatal("could not initialize fee stats")
+			if err = feeWindows.IngestFees(txMeta); err != nil {
+				d.logger.WithError(err).Fatal("could not initialize fee stats: ", d.db.Rollback())
```

Comment:
I would be much cleaner and readable to make the `Rollback` independent of logging